### PR TITLE
Background Mapping - startObserving crash

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Links
N/A

### 🎯 Goal

Fix crash when calling `startObserving` after clearing the databse

### 📝 Summary

We have been observing a crash when accessing objects belonging to an NSManagedObjectContext after removing the persistent store coordinator when using Background Mapping. This could also be seen when not using Background Mapping, but with less frequency.

### 🛠 Implementation

This PR calls `reset()` on the `NSManagedObjectContext`s prior to removing the persistent store coordinator.

### 🧪 Manual Testing Notes

Log In - Log Out dance as fast as possible

Expected result:
- No crash

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [/] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Controllers/DatabaseObserver/DatabaseObserverRemovalListener.swift [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+                // Publish the changes
+                changeAggregator.controllerDidChangeContent(fetchResultsController)
+
+                // Remove delegate so it doesn't get further removal updates
+                frc.delegate = nil
+            }
+
+            if isBackground {
+                context.performAndWait { context.reset() }
```
Comment (martinmitrevski) [inline Sources/StreamChat/Controllers/DatabaseObserver/DatabaseObserverRemovalListener.swift:58]: This seems like the most important change. Can you please explain what was the issue?

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅  An approval from @testableapple would also be needed before merging.

**Review by testableapple — APPROVED** [review]
LGTM 🚢

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**